src/Executor/ScopedContext.php

```diff
@@ -0,0 +1,12 @@
+<?php declare(strict_types=1);
+
+namespace GraphQL\Executor;
+
+/**
+ * This interface can be used to indicate that your context should be scoped.
+ *
+ * A scoped context will only delegate information to children.
+ */
+interface ScopedContext
+{
```

Comment:
I think we should add a `public function clone(): void;` here and use it to describe when exactly it will be called. That will make the mechanics of how this works much more obvious.

Reply:
Done